rpc/rpc.go

```diff
@@ -354,113 +370,139 @@ closeTransport:
 	return nil
 }
 
+func (c *Conn) send(ctx context.Context) error {
+	for {
+		if err := c.sendMessage(ctx, func(msg rpccp.Message) error {
```

Comment:
Style nitpick: given the size of the expression, this would probably scan better as a separate assignment and then if statement, rather than using the compound if form.